core/services/ardupilot_manager/ArduPilotManager.py

```diff
@@ -104,3 +116,31 @@ def start_board(self, boards: List[Tuple[FlightControllerType, str]]) -> bool:
 
     def restart(self) -> None:
         self.mavlink_manager.restart()
+
+    def _load_endpoints(self) -> None:
+        if "endpoints" not in self.configuration:
+            self.configuration["endpoints"] = []
+        else:
+            endpoints = self.configuration["endpoints"]
+            self.mavlink_manager.add_endpoints([Endpoint(e) for e in endpoints])
+
+    def get_endpoints(self) -> List[Endpoint]:
+        return self.mavlink_manager.endpoints()
+
+    def add_endpoint(self, endpoint: Endpoint) -> bool:
+        if self.mavlink_manager.add_endpoint(endpoint):
+            self.configuration.get("endpoints", []).append(endpoint.__dict__)
+            self.settings.save(self.configuration)
+            return True
+        return False
+
+    def remove_endpoint(self, endpoint: Endpoint) -> bool:
+        if self.mavlink_manager.remove_endpoint(endpoint):
+            try:
+                self.configuration.get("endpoints", []).remove(endpoint.__dict__)
+                self.settings.save(self.configuration)
+                return True
+            except Exception as e:
+                warn(f"Error: {e}")
```

Comment:
`Error while removing endpoints: {e}`
Also, try to avoid single letter variables, try `error` or `exception`